pkg/controllers/cloud/configmap_controller.go

```diff
@@ -28,18 +27,59 @@ import (
 
 	"antrea.io/nephe/pkg/cloud-provider/securitygroup"
 	"antrea.io/nephe/pkg/controllers/config"
+	"antrea.io/nephe/pkg/logging"
 )
 
+type ConfigMapHandler func(controllerConfig *config.ControllerConfig)
+
+// controllerConfigType use to identify the fields in ControllerConfig
+type controllerConfigType int
+
 type ConfigMapReconciler struct {
 	client.Client
-	Log              logr.Logger
-	Scheme           *runtime.Scheme
-	ControllerConfig *config.ControllerConfig
-	pendingSyncCount int
+	log    func() logging.Logger
+	Scheme *runtime.Scheme
+
+	ControllerConfig     *config.ControllerConfig
+	pendingSyncCount     int
+	configmapHandlerFunc map[controllerConfigType][]ConfigMapHandler
+}
+
+const (
+	cloudSyncIntervalConfig controllerConfigType = iota
+)
+
+var (
+	configmapReconciler *ConfigMapReconciler
+)
+
+func init() {
+	configmapReconciler = &ConfigMapReconciler{}
+	configmapReconciler.ControllerConfig = &config.ControllerConfig{
+		CloudResourcePrefix: config.DefaultCloudResourcePrefix,
+		CloudSyncInterval:   config.DefaultCloudSyncInterval,
+	}
+	configmapReconciler.configmapHandlerFunc = make(map[controllerConfigType][]ConfigMapHandler)
+	configmapReconciler.pendingSyncCount = 1
+	configmapReconciler.log = func() logging.Logger {
```

Comment:
this seems to be a hack.. please spend sometime with Anand and see if we can come up with a better way

Reply:
Solved.